pkg/services/ngalert/state/manager.go

```diff
@@ -279,16 +284,7 @@ func (st *Manager) ProcessEvalResults(ctx context.Context, evaluatedAt time.Time
 	))
 
 	staleStates := st.deleteStaleStatesFromCache(ctx, logger, evaluatedAt, alertRule)
-	st.deleteAlertStates(tracingCtx, logger, staleStates)
-
-	if len(staleStates) > 0 {
-		span.AddEvent("deleted stale states", trace.WithAttributes(
-			attribute.Int64("state_transitions", int64(len(staleStates))),
-		))
-	}
-
-	st.saveAlertStates(tracingCtx, logger, states...)
-	span.AddEvent("updated database")
+	st.persister.Sync(tracingCtx, span, states, staleStates)
```

Comment:
We can then log a warning when sync fails.

Reply:
This is already done inside the Sync call.